instrumentation/spring-cache-3.1.0/src/main/java/org/springframework/cache/Cache_Instrumentation.java

```diff
@@ -0,0 +1,42 @@
+package org.springframework.cache;
+
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.NewField;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import org.springframework.cache.Cache;
+
+import java.util.logging.Level;
+
+@Weave(type = MatchType.Interface, originalName = "org.springframework.cache.Cache")
+public abstract class Cache_Instrumentation {
+    public Cache.ValueWrapper get(Object key) {
+        Cache.ValueWrapper result = Weaver.callOriginal();
+        NewRelic.incrementCounter("Spring/Cache/" + getProviderClassName() + "/" + getName() + (result == null ? "/Misses" : "/Hits"));
```

Comment:
I'd start the metric name with Cache, then Spring.
This way if we instrument other cache libraries, the metrics will all start with Cache, similar to Datastore or External.